tests/components/bsblan/test_init.py

```diff
@@ -45,3 +47,32 @@ async def test_config_entry_not_ready(
 
     assert len(mock_bsblan.state.mock_calls) == 1
     assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_config_entry_auth_failed_triggers_reauth(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_bsblan: MagicMock,
+) -> None:
+    """Test that BSBLANAuthError during coordinator update triggers reauth flow."""
+    # First, set up the integration successfully
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    # Get the coordinator
+    coordinator = mock_config_entry.runtime_data.coordinator
+
+    # Mock BSBLANAuthError during next update
+    mock_bsblan.initialize.side_effect = BSBLANAuthError("Authentication failed")
+
+    # Manually trigger coordinator update which should raise ConfigEntryAuthFailed
+    try:
+        await coordinator._async_update_data()
```

Comment:
we should not touch internals like runtime data, instead, skip time and then see that a config flow has been created and that the config entry state is in error